### PR TITLE
TEST-#5790: add ASV configs for Dask and Unidist

### LINKED ISSUE #5790: TEST: add ASV configs for Dask and Unidist
These configs allow us to use ASV in the following scenarios (for example):
```bash
asv continuous main-branch custom-branch --launch-method=spawn -b TimeMergeDefault \
   --no-only-changed -a repeat=5 --show-stderr --config asv.conf.unidist.json
```

```bash
asv continuous main-branch custom-branch --launch-method=spawn -b TimeMergeDefault \
   --no-only-changed -a repeat=5 --show-stderr --config asv.conf.dask.json
```

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5790 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: asv_bench/asv.conf.dask.json [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    // timeout in seconds for installing any dependencies in environment
+    // defaults to 10 min
+    //"install_timeout": 600,
+
+    // the base URL to show a commit for the project.
+    "show_commit_url": "https://github.com/modin-project/modin/commit/",
+
+    // The Pythons you'd like to test against.  If not provided, defaults
+    // to the current version of Python used to run `asv`.
+    // "pythons": ["3.7"],
```
Comment (YarShev) [inline asv_bench/asv.conf.dask.json:58]: Modin supports python >=3.8.

File: asv_bench/asv.conf.dask.json [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    // list or empty string indicates to just test against the default
+    // (latest) version. null indicates that the package is to not be
+    // installed. If the package to be tested is only available from
+    // PyPi, and the 'environment_type' is conda, then you can preface
+    // the package name by 'pip+', and the package will be installed via
+    // pip (with all the conda available packages installed first,
+    // followed by the pip installed packages).
+    // "matrix": {
+    //    "pip+ray": ["1.0.1"],
+    //    "pyarrow": ["1.0"]
```
Comment (YarShev) [inline asv_bench/asv.conf.dask.json:75]: These are not Dask engine related. Why do you list it here?

File: asv_bench/asv.conf.dask.json (hunk 1 above)
Comment (anmyachev) [reply asv_bench/asv.conf.dask.json]: Updated

File: asv_bench/asv.conf.dask.json (hunk 2 above)
Comment (anmyachev) [reply asv_bench/asv.conf.dask.json:75]: These are just examples.

**Comment by anmyachev** [comment]
@YarShev anything else to fix? Unidist will not work right now because of how it starts (via `mpiexec -n 1 python`), but config will still be needed. Or I can only add the config for Dask so far.

File: asv_bench/asv.conf.dask.json (hunk 1 above)
Comment (YarShev) [reply asv_bench/asv.conf.dask.json]: We should probably update other config files too as part of a separate issue. Even though it is just a comment.

**Comment by YarShev** [comment]
I think we can merge it. Let's try to find a way to run unidist on MPI with asv in a separate issue.

**Review by YarShev — APPROVED** [review]

File: asv_bench/asv.conf.dask.json (hunk 1 above)
Comment (YarShev) [reply asv_bench/asv.conf.dask.json]: And yes, we should probably rename `asv.conf.json` to `asv.conf.ray.json`.

**Pull request merged**